### ISSUE TITLE
GeometryFixer changes XY coordinates to XYZ coordinates where Z = NaN

### ISSUE THREAD

**Issue by swidz** [NetTopologySuite/NetTopologySuite#567]
After fixing MultiPolygon with `NetTopologySuite.Geometries.Utilities.GeometryFixer.Fix(geometry)` output geometry is Polygon and all coordinates are changed from XY to XYZ where Z = NaN

**Comment by DGuidi** [comment]
I do some investigation and I suspect the problem lies in [this line](https://github.com/NetTopologySuite/NetTopologySuite/blob/c9cf846cdc623f571fcb57c31dbd1c40443a9c38/src/NetTopologySuite/Algorithm/RobustLineIntersector.cs#L155) where a coordonate is force as a CoordinateZ

this is the test I used
```
private class XYCoordinateSequenceFactory : CoordinateSequenceFactory
{
	public XYCoordinateSequenceFactory() : base(Ordinates.XY) { }

	public override CoordinateSequence Create(int size, int dimension, int measures)
	{
		Assert.AreEqual(3, dimension);
		Assert.AreEqual(0, measures);
		return new CoordinateArraySequence(size, 2, measures);
	}
}

[Test]
public void TestIssue567_Geom0()
{
	string wkt;
	using (var strm = Assembly.GetExecutingAssembly().GetManifestResourceStream("NetTopologySuite.Samples.Tests.Github.Issue566and567.wkt"))
	using (var rdr = new StreamReader(strm))
	{
		Assert.IsTrue(string.IsNullOrWhiteSpace(rdr.ReadLine()));
		wkt = rdr.ReadLine();
	}
	Assert.False(string.IsNullOrEmpty(wkt));

	var services = new NtsGeometryServices(new XYCoordinateSequenceFactory());
	var wktRdr = new WKTReader(services);
	var geom = wktRdr.Read(wkt);
	Assert.IsNotNull(geom);
	Assert.IsFalse(geom.IsValid);
	Assert.AreEqual(2, CoordinateArrays.Dimension(geom.Coordinates));

	var geomFixed = GeometryFixer.Fix(geom);
	Assert.IsNotNull(geomFixed);
	Assert.IsTrue(geomFixed.IsValid);
	Assert.AreEqual(2, CoordinateArrays.Dimension(geomFixed.Coordinates));
}
```

actually the test doesn't complete, an exception is thrown instead when computing the fix

**Comment by FObermaier** [comment]
@DGuidi , thanks for you input on this:. I'd prefer to change
https://github.com/NetTopologySuite/NetTopologySuite/blob/d793cd2f8f2ddb47f3f4e3c3d1dd1dd2df514d30/src/NetTopologySuite/Algorithm/IntersectionComputer.cs#L89-L90

to
```C#
return p1.Create(xInt + midx, yInt + midy);
```
Then I'd check in `RobustLineIntersector` if the intersection point can carry a z-ordinate and if it does, interpolate it. That is why I proposed #568.

**Comment by DGuidi** [comment]
@FObermaier `IntersectionComputer` looks not called in the test I've built.
My "patch" checks `LineIntersector.InputLines` to understand if it's possible to handle Z value, but maybe there's a clever way to do this check

**Comment by FObermaier** [comment]
Fixed with a9becc07dcad66fc9405d1869364ac8e6ce8650d
